### ISSUE TITLE
Code annotation numbers overlap in HTML in Safari

### ISSUE THREAD

**Issue by martinfleis** [quarto-dev/quarto-cli#6387]
### Bug description

When adding multiple code annotations to a single Python block in qmd, the HTML output does not work properly in safari, causing overlapping and misaligned numbers.

### Steps to reproduce

````qmd
---
title: "Reproducer"
format: html
---

Introduction to issue

```{python}
a = 1 + 1 # <1>
b = "foo"  # <2>
c = "bar"  # <3>
d = "baz"  # <4>
```
1. It is two.
2. foo
3. bar
4. baz

````

### Expected behavior

It should look like this, when looking at it from Chrome:
<img width="762" alt="Screenshot 2023-07-31 at 20 47 24" src="https://github.com/quarto-dev/quarto-cli/assets/36797143/6232a759-314f-44d2-af4e-3b49827527b2">

### Actual behavior

It looks like this:

![Screenshot 2023-07-31 at 20 47 50](https://github.com/quarto-dev/quarto-cli/assets/36797143/284e7b13-e3d9-4d8b-8ecc-f88cc8bd79f1)

If you zoom in, you see that 1 is under 2. Seems like a CSS issue to me.

### Your environment

- IDE: VSCode (but that is irrelevant I guess)
- OS: macOS Ventura 13.4.1
- Safari 16.5.2

### Quarto check output

```bash
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.6: OK
      Dart Sass version 1.55.0: OK
      Deno version 1.33.4: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.274
      Path: /Applications/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.11 (Conda)
      Path: /Users/martin/mambaforge/envs/stable/bin/python
      Jupyter: 5.3.0
      Kernels: stable, dask_gpd_dev, proceedings, momepy-minimal, geopython_tutorial, python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.1.2
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/library
      knitr: 1.37
      rmarkdown: (None)

      The rmarkdown package is not available in this R installation.
      Install with install.packages("rmarkdown")
```

**Comment by mcanouil** [comment]
I can using Safari 16.5.2.
<img width="831" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/56878f15-c4eb-4bef-82dc-4ad02078cf21">

**Comment by cscheid** [comment]
Yes, I can reproduce on Safari 16.5 as well:

<img width="808" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/9d38eb3e-d27a-4a2e-a014-d6f290633c3a">

Now, the question is whether this is a Safari 16.5 bug or a quarto bug.

**Comment by cscheid** [comment]
This also doesn't happen on latest Firefox:

<img width="844" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/e61ab0e3-3aec-496c-af9c-ffd4f703ed77">

Since the behavior changed from 16.1 to 16.5, I'm a little suspicious of a regression on their side...

**Comment by mcanouil** [comment]
It seems to be related with `position: absolute`.

**Comment by cscheid** [comment]
@mcanouil You're 10 seconds ahead of me :)

**Comment by cscheid** [comment]
If I had to guess, this is a regression on their side introduced at 16.4 https://developer.apple.com/documentation/safari-release-notes/safari-16_4-release-notes. I'm nowhere near a CSS expert, but our CSS isn't particularly sophisticated there.

**Comment by cscheid** [comment]
If you flip the order of each annotation and line of code, you solve the problem on Safari >=16.5, but introduce the reverse problem on Chrome and Firefox...

**Comment by mcanouil** [comment]
I played a bit and I am pretty convince the issue is Safari.

**Comment by cscheid** [comment]
@mcanouil Sure, but we're the ones shipping code that tickles their bug. It'd be nice to avoid it if we can.

**Comment by mcanouil** [comment]
The issue comes from the nested containers.
Although, it's not that simple to isolate, reproduce, in order to report back to Apple.

**Comment by ForBo7** [comment]
I'm on Safari 17.0 and I have the same issue too.

<img width="830" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/85829792/b49f96d6-730c-4c14-96c2-ab484fbf8a07">

**Comment by JD2112** [comment]
Hi!
I have `macOS Sonoma v14.4.1` with `Safari v17.4.1` and the above-mentioned code is working fine for me!
![Screenshot 2024-05-21 at 08 43 12](https://github.com/quarto-dev/quarto-cli/assets/15922543/dd83e3ff-9664-4a1e-ac95-7351e048e38a)

**Comment by mcanouil** [comment]
Per the screenshot, I would not have described this as "working fine". Better yes, but not fine.

**Comment by JD2112** [comment]
> Per the screenshot, I would not have described this as "working fine". Better yes, but not fine.

True, 💯 but I have the same issue with `Firefox v126.0` and `Chrome v125.0` on same `macOS v14.4.1`
<img width="1647" alt="Screenshot 2024-05-21 at 09 13 46" src="https://github.com/quarto-dev/quarto-cli/assets/15922543/83f7cd55-b782-41e6-bf05-17093e292e1e">
